lnwallet/musig_session.go

```diff
@@ -0,0 +1,535 @@
+package lnwallet
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr/musig2"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/shachain"
+)
+
+// MusigPartialSig is a wrapper around the base musig2.PartialSignature type
+// that also includes information about the set of nonces used, and also the
+// signer. This allows us to implement the input.Signature interface, as that
+// requires the ability to perform abstract verification based on a public key.
+type MusigPartialSig struct {
+	// sig is the actual musig2 partial signature.
+	sig *musig2.PartialSignature
+
+	// signerNonce is the nonce used by the signer to generate the partial
+	// signature.
+	signerNonce [musig2.PubNonceSize]byte
```

Comment:
Could it be useful to define a type for `[musig2.PubNonceSize]byte`, to also make it a bit stricter?

Reply:
You mean something like:
```go
type MusigNonce [musig2]PubNonceSize]byte
```
?

We actually have that in `lnwire` as `lnwire.Musig2Nonce`, so I'll slot that in here.